tools/dependency/release_dates.py

```diff
@@ -48,22 +56,51 @@ def VerifyAndPrintLatestRelease(dep, repo, metadata_version, release_date):
 def VerifyAndPrintReleaseDate(dep, github_release_date, metadata_release_date):
   mismatch = ''
   iso_release_date = FormatUtcDate(github_release_date)
-  print(f'{dep} has a GitHub release date {iso_release_date}')
+  Colorize("GREEN", f'{dep} has a GitHub release date {iso_release_date}')
   if iso_release_date != metadata_release_date:
     raise ReleaseDateError(f'Mismatch with metadata release date of {metadata_release_date}')
 
 
 # Extract release date from GitHub API.
-def GetReleaseDate(repo, metadata_version, github_release):
+def GetReleaseDate(dep, repo, metadata_version, github_release):
   if github_release.tagged:
-    tags = repo.get_tags()
-    for tag in tags:
-      if tag.name == github_release.version:
-        return tag.commit.commit.committer.date
+
+    try:
+      latest = repo.get_latest_release()
+    except github.UnknownObjectException:
+      latest = ""
+
+    if latest and (github_release.version <= latest.tag_name):
```

Comment:
Nit: gratuitous parens

Reply:
`github_release.version <= latest.tag_name` <-- is this lexicographic comparison always safe? E.g. `1.2.3 < 10.2.3` but `1_2_3 > 10_2_3`.

Reply:
Can you add a comment on the logic here. Are you saying that if there is an earlier latest release than last tag, we should basically ignore the tag?

Reply:
The comparison were working but I've added the `version.parse` wrapper for additional safety. I've added some comments to explain the logic. There are some edge cases where repos have tagged releases that are not marked latest for example. In the new version warnings I've explained that we exclude release candidates with Kafka being the edge case